convert2rhel_insights_tasks/main.py

```diff
@@ -452,10 +471,15 @@ def archive_report_file(file):
     shutil.move(file, archive_log_file)
 
 
-def gather_json_report(report_file):
+def gather_json_report(report_file, returncode):
     """Collect the json report generated by convert2rhel."""
     logger.info("Collecting JSON report.")
 
+    # Do not gather the report when rollback failed. In this case the report isn't generated since
+    # it would be useless - the system is in undefinded state.
+    if returncode == 1 or returncode is None:
+        return {}
```

Comment:
Can conversion (not analysis) end with exit code 1 and generate report? this would cause missing report data in that case

Reply:
Yes, you are right. I missed this situation, I thought the exit code 1 can be only when the rollback fails... I will fix it

Thanks!